### PR TITLE
refactor(internal/carver): try to improve readability a bit

### PR DESCRIPTION
**Pull request by codyoss**
Some calculations were done in multiple places. Decided to make
modInfo more useful and pass those structs around.

Also included a fix to update snippets mod so builds pass.

### CODE REVIEW

**Comment by codyoss** [comment]
Sample gen: https://github.com/googleapis/google-cloud-go/pull/4526

File: internal/carver/cmd/main.go [hunk 1]
```diff
@@ -97,84 +97,115 @@ func main() {
...
 }
 
 type modInfo struct {
-	filePath   string
-	moduleName string
-	tag        string
+	// path is the filepath to the module locally.
+	path string
+	// name is the import path of the module.
+	name string
```
Comment (tbpg) [inline internal/carver/cmd/main.go:123]: Consider `importPath`.

File: internal/carver/cmd/main_test.go [hunk 2]
```diff
@@ -97,22 +96,17 @@ func TestParsePkgName(t *testing.T) {
 			want: "cx",
 		},
 		{
-			name:    "invalid",
-			in:      "cloud.google.com",
-			wantErr: true,
+			name: "invalid",
```
Comment (tbpg) [inline internal/carver/cmd/main_test.go:99]: Seems like this is now valid?

File: internal/carver/cmd/main_test.go [hunk 3]
```diff
@@ -97,22 +96,17 @@ func TestParsePkgName(t *testing.T) {
...
-				t.Fatalf("parsePkgName(%q) = nil, want error", tt.in)
-			}
-			if !tt.wantErr && err != nil {
-				t.Fatalf("parsePkgName(%q) = %v, wantErr false", tt.in, err)
-			}
+			mi := &modInfo{name: tt.in}
+			got := mi.PkgName()
 			if got != tt.want {
-				t.Fatalf("parsePkgName(%q) = %q, want %q", tt.in, got, tt.want)
+				t.Fatalf("PkgName() = %q, want %q", got, tt.want)
```
Comment (tbpg) [inline internal/carver/cmd/main_test.go:109]: Consider still including the `name` as that is helpful for debugging.

File: internal/carver/cmd/main_test.go (hunk 2 above)
Comment (codyoss) [reply internal/carver/cmd/main_test.go]: Oops, yup!

File: internal/carver/cmd/main_test.go (hunk 3 above)
Comment (codyoss) [reply internal/carver/cmd/main_test.go]: Done.

File: internal/carver/cmd/main.go (hunk 1 above)
Comment (codyoss) [reply internal/carver/cmd/main.go]: Done

**Pull request merged**